dashboard/src/utils/explorer.ts

```
@@ -111,9 +121,30 @@ export const explorerTx = (chainId: ChainId, tx: string) =>
     ? `https://arbiscan.io/tx/${tx}`
     : chainId === CHAIN_ID_INJECTIVE
     ? `https://explorer.injective.network/transaction/${tx}`
+    : chainId === CHAIN_ID_SUI
+    ? `https://suiexplorer.com/txblock/${tx}`
     : '';
 
 export const explorerVaa = (key: string) =>
   `https://wormhole.com/explorer/?emitterChain=${key.split('/')[0]}&emitterAddress=${
     key.split('/')[1]
   }&sequence=${key.split('/')[2]}`;
+
+export const getExplorerTxHash = (chain: ChainId, txHash: string) => {
+  let explorerTxHash = '';
+  if (isCosmWasmChain(chain)) {
+    explorerTxHash = txHash.slice(2);
+  } else if (chain === CHAIN_ID_SUI) {
+    const txHashBytes = Buffer.from(txHash.slice(2), 'hex');
+    explorerTxHash = base58.encode(txHashBytes);
+  } else if (!isEVMChain(chain)) {
+    try {
+      explorerTxHash = tryHexToNativeString(txHash.slice(2), CHAIN_INFO_MAP[chain].chainId);
+    } catch (e) {
+      return txHash;
```

If this fails it is better to return the input instead of empty string?